agent/host.go

```diff
@@ -52,6 +54,59 @@ type CPUInfo struct {
 	Microcode  string `json:"Microcode,omitempty"`
 }
 
+// Extract class and ID from the children node
+func extractClassAndID(children map[string]interface{}) (string, string) {
+	var foundID string
+	var foundClass string
+	for objectName, objectValue := range children {
+		switch objectValue.(type) {
+		case string:
+			if objectName == "class" {
+				foundClass = objectValue.(string)
+			}
+			if objectName == "id" {
+				foundID = objectValue.(string)
+			}
+			// Once we discover the class and ID, let's return them
+			if (len(foundID) > 0) && (len(foundClass) > 0) {
+				return foundClass, foundID
+			}
+		}
+	}
+	return "", ""
+}
+
+// Parse the LSHW data structure to rename children nodes
+func parseLshw(items map[string]interface{}) {
+	// For each object of the LSHW structure
+	for objectName, objectValue := range items {
+		switch objectValue.(type) {
+		case []interface{}:
+			// Only consider object that have children
+			if objectName != "children" {
+				return
+			}
+			// When we have children
+			for _, child := range objectValue.([]interface{}) {
+				// For each child
+				class, ID := extractClassAndID(child.(map[string]interface{}))
```

Comment:
As there is no guaranty for the lshw output I think we should check all the type assertion like

```
childType, ok :=  child.(map[string]interface{})
if ok {
```

Reply:
Yup, good catch. Fixed.